core/new-gui/src/app/dashboard/component/feature-container/user-project-list/user-project-list.component.html

```diff
@@ -162,11 +179,12 @@ <h2>Projects</h2>
                   nz-input
                   placeholder="Enter project description"
                   nzAutosize
-                  maxlength="500"
+                  maxlength="10000"
```

Comment:
be careful with the maxlength limit. here you are using 10k for both frontend and db constrain. sometimes frontend limits a number but when received and by backend, it could append some characters like "\n", so that when it inserts to db it will still fail the size constrain of db. you can test it with a small number to see if this handles correctly, otherwise we could have a smaller limit on the frontend than db.

Reply:
use the constant you had `MAX_PROJECT_DESCRIPTION_CHAR_COUNT`

Reply:
I updated to use the constant and tested that I can successfully insert a 10K length string with no errors.

Reply:
I did run into another issue though with how characters are counted in textarea across different browsers.  

Do we currently support Texera for Safari, or is it only Chrome?

It seems certain browsers like Safari (I'm using version 15.6.1) counts the `Enter` key a 2 character carriage return \r\n, but other browsers like Chrome count it as 1 character \n.  

This can result in inconsistencies with the displayed character count like below in Safari, where `textarea` stops the user from typing past 7 characters because it thinks 10 has already been reached.
<img width="310" alt="Screen Shot 2023-01-22 at 19 00 03" src="https://user-images.githubusercontent.com/57075492/213958907-a805ca5b-9d7b-415b-a1da-50059bb28305.png">

I can look into this for another PR but wanted to get your opinion on this first

Reply:
we do need to support safari.

however, `\r\n` is a single character and should be counted only once. are we calculating the number of characters ourselves? 

To solve it, we can either find a solution to make sure characters like `\r\n` is counted as 1 character, or we can find a way to convert all `\r\n` to `\n`, as long as it displays correctly when rendered. Either way, this is a minor issue, if you do not want to include the fix in this PR, please track it with an issue.

Reply:
Sounds good, filed as a new issue with link to this comment